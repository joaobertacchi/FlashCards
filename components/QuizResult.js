// @flow
import React from 'react';
import { Text, View, StyleSheet } from 'react-native';
import type { Questions, ScoreType } from '../types';

import Button from './Button';

type Props = {
  questions: Questions,
  score: ScoreType,
  onRestart: () => void,
  onBackToDeck: () => void,
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'space-around',
  },
  score: {
    textAlign: 'center',
    fontSize: 32,
  },
});

const QuizResult = ({
  questions, score, onRestart, onBackToDeck,
}: Props) => {
  const total = questions.length;
  const correct = Object.values(score).reduce(
    (acc, isCorrect) => (isCorrect === true ? acc + 1 : acc),
    0,
  );
  return (
    <View style={styles.container}>
      <Text style={styles.score}>{`You scored ${Math.round((100 * correct) / total)}%`}</Text>
      <Button onPress={onRestart} text="Restart Quiz" type="success" />
      <Button onPress={onBackToDeck} text="Back to Deck" type="primaryLight" />
    </View>
  );
};

export default QuizResult;
