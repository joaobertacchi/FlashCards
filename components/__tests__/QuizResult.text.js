import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import QuizResult from '../QuizResult';

describe('CardBack snapshot', () => {
  jest.useFakeTimers();
  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it('renders correctly', () => {
    const questions = [
      {
        question: 'A',
        answer: 'B',
      },
    ];
    const score = {
      0: true,
    };
    const tree = renderer.create(<QuizResult questions={questions} score={score} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const questions = [
      {
        question: 'A',
        answer: 'B',
      },
    ];
    const score = {
      0: false,
    };
    const tree = renderer.create(<QuizResult questions={questions} score={score} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const questions = [
      {
        question: 'A',
        answer: 'B',
      },
    ];
    const score = {
      0: undefined,
    };
    const tree = renderer.create(<QuizResult questions={questions} score={score} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
