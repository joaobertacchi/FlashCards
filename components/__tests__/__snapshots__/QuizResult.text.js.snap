// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBack snapshot renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-around",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 32,
        "textAlign": "center",
      }
    }
  >
    You scored 100%
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 20,
          "marginHorizontal": 20,
          "marginTop": 10,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "green",
          "borderRadius": 7,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 12,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 18,
              "minWidth": 180,
              "textAlign": "center",
            },
            Object {
              "minWidth": undefined,
            },
          ]
        }
      >
        Restart Quiz
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 20,
          "marginHorizontal": 20,
          "marginTop": 10,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 7,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 12,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2f95dc",
              "fontSize": 18,
              "minWidth": 180,
              "textAlign": "center",
            },
            Object {
              "minWidth": undefined,
            },
          ]
        }
      >
        Back to Deck
      </Text>
    </View>
  </View>
</View>
`;

exports[`CardBack snapshot renders correctly 2`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-around",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 32,
        "textAlign": "center",
      }
    }
  >
    You scored 0%
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 20,
          "marginHorizontal": 20,
          "marginTop": 10,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "green",
          "borderRadius": 7,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 12,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 18,
              "minWidth": 180,
              "textAlign": "center",
            },
            Object {
              "minWidth": undefined,
            },
          ]
        }
      >
        Restart Quiz
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 20,
          "marginHorizontal": 20,
          "marginTop": 10,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 7,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 12,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2f95dc",
              "fontSize": 18,
              "minWidth": 180,
              "textAlign": "center",
            },
            Object {
              "minWidth": undefined,
            },
          ]
        }
      >
        Back to Deck
      </Text>
    </View>
  </View>
</View>
`;

exports[`CardBack snapshot renders correctly 3`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-around",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 32,
        "textAlign": "center",
      }
    }
  >
    You scored 0%
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 20,
          "marginHorizontal": 20,
          "marginTop": 10,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "green",
          "borderRadius": 7,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 12,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 18,
              "minWidth": 180,
              "textAlign": "center",
            },
            Object {
              "minWidth": undefined,
            },
          ]
        }
      >
        Restart Quiz
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 20,
          "marginHorizontal": 20,
          "marginTop": 10,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 7,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 12,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2f95dc",
              "fontSize": 18,
              "minWidth": 180,
              "textAlign": "center",
            },
            Object {
              "minWidth": undefined,
            },
          ]
        }
      >
        Back to Deck
      </Text>
    </View>
  </View>
</View>
`;
